engine/src/main/scala/cromwell/engine/workflow/lifecycle/MaterializeWorkflowDescriptorActor.scala

```diff
@@ -429,8 +429,9 @@ class MaterializeWorkflowDescriptorActor(serviceRegistryActor: ActorRef,
       for {
         wf <- CwlDecoder.decodeAllCwl(cwlFile).map {
           _.select[Workflow].get
+          //TODO move this unsafeRunSync to the fringe so as to leave more power to the caller
         }.value.unsafeRunSync.toValidated
-        executable <- wf.womExecutable
+        executable <- wf.womExecutable.toValidated
```

Comment:
oh I would have thought this would have gone to `Checked` to take advantage of its real `flatMap`

Reply:
it should be, it should really be a `Parse` like decodeAllCwl.  

I was lazy and wrote a TODO.  am doing the hard work now.